tests/kernel/kirk.pm

```diff
@@ -0,0 +1,70 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Executes kirk testing framework
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+use base 'opensusebasetest';
+use testapi qw(get_var get_required_var);
+use utils;
+use strict;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+
+our $result_file = 'result.json';
+
+sub run
+{
+    my ($self) = @_;
+    my $repo = get_var('KIRK_REPO', 'https://github.com/acerv/kirk.git');
```

Comment:
In general this looks good. Kirk should really be in a package, but I would not volunteer to do that.